cmd/internal/backup/providers/local/local.go

```diff
@@ -118,8 +143,6 @@ func (b *BackupProviderLocal) UploadBackup(_ context.Context, sourcePath string)
 // GetNextBackupName returns a name for the next backup archive that is going to be uploaded
 func (b *BackupProviderLocal) GetNextBackupName(_ context.Context) string {
 	name := strconv.FormatInt(b.nextBackupCount, 10)
-	b.nextBackupCount++
-	b.nextBackupCount = b.nextBackupCount % b.config.ObjectsToKeep
 	return name
```

Comment:
I think that moving these lines contradicts with what this public function is promising? At the same time I also don't like this interface function very much, I think we can probably drop it in the next PR when you enable the compressor to handle streams. 

For now, I would suggest to keep the lines here in order to maintain old behavior and store the current name for the backup in the provider struct in order to use it in `UploadBackup`.